### DISCUSSION TITLE
SoundExchange Reports for LIVE Stream Connections

### DISCUSSION THREAD

**Discussion by slyderc** [AzuraCast/AzuraCast#6689] [Q&A]
Greetings!  Is anyone else using live DJ streaming connections to air "live" broadcasts?  We typically run most everything this way and are diligent about including song metadata for proper SoundExchange reporting.  It doesn't appear that these live shows are included in the reports, however, and only include locally-uploaded MP3 files that get played from the AutoDJ feature.  Is this a possibility to add for a future release?  Thank you!

**Comment by Moonbase59** [comment]
Sorry to say, but that will probably stay wishful thinking for another few years. We’ve been looking for something since more than 20 years, and it is almost impossible. Not just with AzuraCast, but with _any_ streaming solution.

Reason being: There is a plethora of streaming software (IDJC, SAM, mixxx, even Butt), and what you can get in a DJ’s _stream_ is at most a small little text field that contains something like "Artist - Title". So if the DJ _client_ can’t generate it (_very_ few can), you’ll be lost here. And try getting data from a DJ in real life…

The best "solution" (still no report, no ISRCs) we’ve come up with in 20+ years is:
- DJs _must_ stream metadata (so we at least get "Artist - Title") and are _forced_ to only broadcast titles with good metadata (we have procedure for that, hint: use MusicBrainz Picard for tagging).
- The stations always make automated _recordings_ of DJ streams (also for legal purposes) in _MP3 + CUE file_ format.
- I wrote a little cue file parser long ago that parses the cue file and makes a human-readable _list_ (text file) out of it. This was originally intended to be uploaded by the the DJ after their show, and actually all DJs that use _IDJC_ for their broadcasts _must_.
- One could think of automating that and generate it on the server, if it creates CUE files with date and time for the recordings.
- Still no Sound Exchange format, or ISRCs. But at least artist and title. Won’t get better, I fear.

Here’s an old one I just grabbed randomly:

```
Recorded with IDJC - [2019-06-21][22:45:58]
aufzeichnung.2019-06-21.224558-r01.mp3
22:45 - 01. Joan Jett and the Blackhearts - A.C.D.C. (2006) (04:47)
22:50 - 02. Airbourne - Runnin' Wild (2007) (03:38)
22:54 - 03. Sammy Hagar - I'm on a Roll (2008) (02:39)
22:57 - 04. Judas Priest - Riding on the Wind (2009) (03:28)
23:00 - 05. Radio Paranoid - Rock und Metal! (00:04)
23:00 - 06. H.E.A.T - Tonight (2010) (03:12)
23:03 - 07. Los Bastardos Finlandeses - Crazy (2011) (02:58)
23:06 - 08. Nickelback - When We Stand Together (2012) (03:09)
23:09 - 09. 42 Decibel - Smokin' Fire (2013) (02:51)
23:12 - 10. Dynamite - Hail Rock 'n' Roll (2014) (03:18)
23:16 - 11. Radio Paranoid - More Music - More Variety - Less Talk (00:07)
23:16 - 12. Razzmattazz - Down on My Knees (2015) (03:15)
23:19 - 13. Deck Janiels - Fuck 'em All (2016) (02:58)
23:22 - 14. Ohrenfeindt - Deine Mudder singt bei Lordi (2017) (03:18)
23:25 - 15. Voodoo Circle - Unknown Stranger (2018) (03:22)
23:29 - 16. Moonbase - Das war's! ()
```

As you can see, not much (meant for humans), but it has date and playout time, sequence number and artist/title/playing duration.

The CUE file IDJC generated looks like this (a typical IDJC auto-recording CUE file):

```
TITLE "[2019-06-21][22:45:58]"
PERFORMER "Recorded with IDJC"
FILE "aufzeichnung.2019-06-21.224558-r01.mp3" MP3
  TRACK 01 AUDIO
    TITLE "A.C.D.C. (2006)"
    PERFORMER "Joan Jett and the Blackhearts"
    REM ALBUM "Sinner"
    INDEX 01 00:00:00
  TRACK 02 AUDIO
    TITLE "Runnin' Wild (2007)"
    PERFORMER "Airbourne"
    REM ALBUM "Runnin' Wild"
    INDEX 01 04:47:28
  TRACK 03 AUDIO
    TITLE "I'm on a Roll (2008)"
    PERFORMER "Sammy Hagar"
    REM ALBUM "Cosmic Universal Fashion"
    INDEX 01 08:25:32
  TRACK 04 AUDIO
    TITLE "Riding on the Wind (2009)"
    PERFORMER "Judas Priest"
    REM ALBUM "A Touch of Evil"
    INDEX 01 11:04:43
  TRACK 05 AUDIO
    TITLE "Rock und Metal!"
    PERFORMER "Radio Paranoid"
    INDEX 01 14:32:68
  TRACK 06 AUDIO
    TITLE "Tonight (2010)"
    PERFORMER "H.E.A.T"
    REM ALBUM "Freedom Rock"
    INDEX 01 14:37:10
  TRACK 07 AUDIO
    TITLE "Crazy (2011)"
    PERFORMER "Los Bastardos Finlandeses"
    REM ALBUM "Saved by Rock 'n' Roll"
    INDEX 01 17:49:74
  TRACK 08 AUDIO
    TITLE "When We Stand Together (2012)"
    PERFORMER "Nickelback"
    INDEX 01 20:48:17
  TRACK 09 AUDIO
    TITLE "Smokin' Fire (2013)"
    PERFORMER "42 Decibel"
    REM ALBUM "Hard Rock 'n' Roll"
    INDEX 01 23:57:26
  TRACK 10 AUDIO
    TITLE "Hail Rock 'n' Roll (2014)"
    PERFORMER "Dynamite"
    REM ALBUM "Blackout Station"
    INDEX 01 26:48:49
  TRACK 11 AUDIO
    TITLE "More Music - More Variety - Less Talk"
    PERFORMER "Radio Paranoid"
    INDEX 01 30:06:56
  TRACK 12 AUDIO
    TITLE "Down on My Knees (2015)"
    PERFORMER "Razzmattazz"
    REM ALBUM "Sons of Guns"
    INDEX 01 30:14:44
  TRACK 13 AUDIO
    TITLE "Fuck 'em All (2016)"
    PERFORMER "Deck Janiels"
    REM ALBUM "II"
    INDEX 01 33:30:21
  TRACK 14 AUDIO
    TITLE "Deine Mudder singt bei Lordi (2017)"
    PERFORMER "Ohrenfeindt"
    REM ALBUM "Zwei Fäuste für Rock 'n' Roll"
    INDEX 01 36:28:69
  TRACK 15 AUDIO
    TITLE "Unknown Stranger (2018)"
    PERFORMER "Voodoo Circle"
    REM ALBUM "Raised On Rock"
    INDEX 01 39:47:39
  TRACK 16 AUDIO
    TITLE "Das war's!"
    PERFORMER "Moonbase"
    INDEX 01 43:09:63
```

Since this comes up again and again, I’d be interested in how others do it, years later.

**Reply by slyderc** [reply]
Thanks for the details.  If the metadata is being sent as part of a live broadcast, Azuracast/Icecast is capturing it just fine and displaying it in the embedded player and play history, for example.  What isn't happening is having this metadata logged for use in the AzuraCast built-in SoundExchange reporting.  That's the part I'm hoping for, in addition to the automated ISRC lookups that's advertised with the radio button on that report page.  :)

Otherwise, we're using the excellent Radiologik playout software for MacOS which has excellent automation/scripting tools.  I'm creating a JSON file with all the currently-playing metadata and logging it locally with some custom Python stuff, but it's nowhere near as elegant as what it looks like AzuraCast could create with its report builder.
